### ISSUE TITLE
https://github.com/erlang/otp/pull/6312

### ISSUE THREAD

**Issue by dgud** [eksperimental/maintenance#1]
Tests are failing for [PR-6312](https://github.com/erlang/otp/pull/6312)

I have updated the functionality in OTP-25, and added another file from the specs to the tests dir,
I didn't update the instructions since I was stupid and did not think it should be affected by new specs.

Without looking at it I think that is the problem, can you take a look?
Or else I will take a look later when I have time.

**Comment by eksperimental** [comment]
> I have updated the functionality in OTP-25, and added another file from the specs to the tests dir, I didn't update the instructions since I was stupid and did not think it should be affected by new specs.

Hi @dgud,
do you have the PR or the commit that made these changes, so I can have a look at

**Comment by dgud** [comment]
merge of 'dgud/stdlib/unicode/add-categories/OTP-18150'     => bb266f56661eced6104666f65f363edbfa5fb492

The bin file is generated from out commented code in gen_unicode_mod script..
Might have been a bad idea though.

+    %% Export testfile
+    %% Dict1 = lists:map(fun({Id,{CCC, Canon, Compat, Cat}}) ->
+    %%                           {_, ECat} = lists:keyfind(Cat, 1, category_translate()),
+    %%                           {Id, {CCC, Canon, Compat, ECat}}
+    %%                   end, Dict0),
+    %% file:write_file("../test/unicode_util_SUITE_data/unicode_table.bin", term_to_binary(Dict1, [compressed])),

**Comment by eksperimental** [comment]
Yes, it definitely needs to be updated at least to include the updated version of `lib/stdlib/uc_spec/EastAsianWidth.txt`

Regarding the bin file, wouldn't it be better to let the test suite generate it on the fly?

**Comment by eksperimental** [comment]
Oh.. I see that file is solely being used for the tests. But if we are going to generated it automatically, would there be any point in doing that test?

**Comment by dgud** [comment]
I have to look, don't even remember what I did there, pre-vacation stuff.

The bin-file is used to verify that I have generated code for every codepoint,
to minimize size I only generated code whenever they differ from the default case,
or the info can't be found in other tables.

So I think we need the test, and we can't generate it from the tests because we test on installed
releases and the uc_spec dir is not included in the installed files.

**Comment by eksperimental** [comment]
I will look into generating the file then and including the new .txt file.

Thank you.

**Comment by dgud** [comment]
Maybe as an option to the escript which generates the test file.
So you can run the script to generate it.

**Comment by eksperimental** [comment]
Hi @dgud, I think we need the script to generate the .bin file. Do you think you could take care of that? Later I could port it to Elixir and include it in this app.

**Comment by eksperimental** [comment]
Actually, as you mentioned, if the escript can accept an option to generate the file, that would suffice.

**Comment by dgud** [comment]
I'll try to remember to fix it, been busy with other stuff.
Ping me here if I forget.

**Comment by dgud** [comment]
I have totally forgot about this, and 26 is on the way.. 
See https://github.com/erlang/otp/pull/6961
If you want I can merge this or you can incorporate the changes into your PR, I don't care.
1/2 a year to uncomment 4 lines of code that is not the highlight of my career :-/

**Comment by eksperimental** [comment]
Hi, 
I think it will be better if you incorporate the changes in the PR, then I will update my app accordingly to work for the next Unicode version.

**Comment by eksperimental** [comment]
Hi @dgud, a new version of Unicode has come out this months and I would like to update the app so it can work with Erlang/OTP. Could you please be so kind to give me an updated of what changes were required to do from that original PR, so I can update the code accordingly. Thank you.

**Comment by dgud** [comment]
I have added a step 6 to the instructions :-) 


6. Check if the test file needs to be updated:
   (cd $ERL_TOP/lib/stdlib/uc_spec; escript gen_unicode_mod.escript update_tests)
   If ../test/unicode_util_SUITE_data/unicode_table.bin is updated include it in
   the commit.
